packages/@lwc/engine-core/src/libs/mutation-tracker/index.ts

```diff
@@ -64,6 +65,34 @@ export function valueObserved(target: object, key: PropertyKey) {
     }
 }
 
+// Putting this here for now, the idea is to subscribe to a signal when there is an active template reactive observer.
+// This would indicate that:
+//      1. The template is currently being rendered
+//      2. There was a call to a getter bound to the LWC class
+// With this information we can infer that it is safe to subscribe the re-render callback to the signal, which will
+// mark the VM as dirty and schedule rehydration.
+// The onUpdate function here mirrors the one used in the template reactive observer, which is to mark the
+// VM as dirty and re-render.
+// In a future optimization, rather than re-render the entire VM we could use fine grained reactivity here
+// to only re-render the part of the DOM that has been changed by the signal.
+// jtu-todo: find a better place to put this that doesn't require mutation-tracker to know about the specifics of signals
+export function signalValueObserved(
+    s: Signal<any>,
+    onUpdate: OnUpdate,
+    unsubscribe: Array<Unsubscribe>
+) {
+    if (currentReactiveObserver === null) {
+        return;
+    }
```

Comment:
I'm using this as a temporary way to check if the template is currently rendering (template reactive observer is active).

Plan to move the logic into a cleaner abstraction.